src/backend/access/appendonly/aomd.c

```diff
@@ -250,6 +251,10 @@ mdunlink_ao_perFile(const int segno, void *ctx)
 	char *segPathSuffixPosition = unlinkFiles->segpathSuffixPosition;
 
 	sprintf(segPathSuffixPosition, ".%u", segno);
+
+	if (do_truncate(segPath) < 0 && errno == ENOENT)
```

Comment:
May be add some comments before, like in other places?

Reply:
done